stow.go

```diff
@@ -145,6 +145,14 @@ func Register(kind string, makefn func(Config) (Location, error), kindmatchfn fu
 	lock.Lock()
 	defer lock.Unlock()
 	locations[kind] = makefn
+
+	// if already registered, leave
+	for _, k := range kinds {
```

Comment:
we need to check this before setting `locations`.

What if we do `if _, ok := locations[kind]; ok { return }`?